LjinuxRoot/bin/jpkg.py

```diff
@@ -0,0 +1,184 @@
+            errored = True
+            break
+        fl.append(extpath)
+        del extpath, fileext
+
+    # backup path
+    bckdir = getcwd()
+
+    updatee = list()
+
+    # parsing
+    if not errored:
+        for fileext in fl:
+            print(f'JPKG: Reading properties "{fileext}" ...')
+            chdir("/LjinuxRoot" + fileext)
+
+            if "Manifest.json" in listdir():
+                manifest = None
+                with open("Manifest.json", "r") as manifest_f:
+                    try:
+                        manifest = json.load(manifest_f)
+                        if manifest["package_name"] not in pklist[0].keys():
+                            pklist[0].update(
+                                {
+                                    manifest["package_name"]: [
+                                        manifest["version"],
+                                        manifest["dependencies"],
+                                        manifest["conflicts"],
+                                    ]
+                                }
+                            )
+                            if manifest["JPKG_minimum_version"] > jpkg_version:
+                                errored = True
+                                print(
+                                    "JPKG Error: Package "
+                                    + manifest["package_name"]
+                                    + " requires a higher version of jpkg, cannot continue."
+                                )
+                        else:
+                            vrs = pklist[0][manifest["package_name"]][0]
+                            if (
+                                vrs[0] > manifest["version"][0]
+                                or vrs[1] > manifest["version"][1]
+                                or vrs[2] > manifest["version"][2]
+                            ):  # update
+                                updatee.append(manifest["package_name"])
+                                pklist[0][manifest["package_name"]] = [
+                                    manifest["version"],
+                                    manifest["dependencies"],
+                                    manifest["conflicts"],
+                                ]
+                            else:  # version <=
+                                errored = True
+                                print(
+                                    "JPKG Error: Package "
+                                    + manifest["package_name"]
+                                    + " version smaller or equal to current."
+                                )
+                            del vrs
+
+                        for depsc in manifest["dependencies"]:
+                            if depsc not in pklist[1]:
+                                pklist[1].append(depsc)
+                            del depsc
+                        for confc in manifest["conflicts"]:
```

Comment:
pklist[1].extend = list(filter(lambda depsc: depsc not in pklist[1], manifest["dependencies"]))

Try this, if it works its much faster and less verbose, please run how to use lamdas.

Reply:
Yes master pymele